bee-vote/src/fpc.rs

```diff
@@ -512,28 +505,31 @@ where
         let mut queried_opinions = QueriedOpinions {
             opinion_giver_id: opinion_giver.id().to_string(),
             opinions: HashMap::new(),
-            times_counted: selected_count,
+            times_counted: opinion_select_count,
         };
 
         let mut vote_map_guard = vote_map.write().await;
 
         // Get opinions on a voting object and add to vote map.
-        let mut query_voting_objects = |objects: &Vec<VoteObject>| {
-            for (i, object) in objects.iter().enumerate() {
+        let mut query_voting_objects = |vote_objects: &Vec<VoteObject>| {
+            for (vote_object, opinion) in vote_objects
+                .iter()
+                .zip(opinions.iter()) 
+            {
                 let mut votes = vote_map_guard
-                    .get(object)
-                    .map_or(Opinions::new(vec![]), |opt| opt.clone());
+                    .get(vote_object)
+                    .map_or_else(|| Opinions::new(vec![]), |votes| votes.clone());
```

Comment:
I'd reserve the capacity to hold `opinion_select_count` items in the vector upfront.